### PR TITLE
Add movement-related actions

### PR DESCRIPTION
**Pull request by kaspermelheim**
Implemented movement-related actions, JUnit tests and javadocs/comments in the code. Also formatted and cleaned up the code.

### CODE REVIEW

File: src/main/java/inf112/isolasjonsteamet/roborally/actions/CheckPoint.java [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+import inf112.isolasjonsteamet.roborally.board.Board;
+import inf112.isolasjonsteamet.roborally.players.Player;
+
 /**
  * Checkpoint is an action that let's us keep track of when a player has reached a chekcpoint.
- *
  */
 public class CheckPoint implements Action {
 
+	private int id;
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/actions/CheckPoint.java]: Let's wait with adding fields here until we know how we'll use them

File: src/main/java/inf112/isolasjonsteamet/roborally/app/Game.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+import inf112.isolasjonsteamet.roborally.players.PlayerImpl;
+import inf112.isolasjonsteamet.roborally.util.Coordinate;
+import inf112.isolasjonsteamet.roborally.util.Orientation;
+
+/**
+ * Game class that starts a new game.
+ */
+public class Game extends InputAdapter implements ApplicationListener {
+
+	private SpriteBatch batch;
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/app/Game.java]: Can we make all the variables here private?

File: src/main/java/inf112/isolasjonsteamet/roborally/board/Board.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ */
+public interface Board {
+
+	/**
+	 * Get a list of the Players on the Board.
+	 *
+	 * @return playerList
+	 */
+	List<Player> getPlayers();
+
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/board/Board.java:20]: Remove stray newline here

File: src/main/java/inf112/isolasjonsteamet/roborally/board/BoardClientImpl.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		}
+
+		return ImmutableList.copyOf(accX);
+	}
+
+	/**
+	 * Update the playerview.
+	 */
+	public void updatePlayerView() {
+		for (Player player : players) {
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/board/BoardClientImpl.java:111]: This method won't work as well if we move the player start position in the map file. Can we remove the player from the map file?

File: src/main/java/inf112/isolasjonsteamet/roborally/board/BoardImpl.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+		if (tilesX.size() <= pos.getY()) {
+			return ImmutableList.of(Tiles.HOLE);
+		}
+
+		return tilesX.get(pos.getY());
+	}
+
+	/**
+	 * Check if x and y has valid values.
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/board/BoardImpl.java]: Better description for this is probably something like "Check if the board is in a valid state"

**Review by Katrix — CHANGES_REQUESTED** [review]
I'd say remove the javadoc from child classes, where the superclass (or interface) already has comments for that thing. If Checkstyle complains. Use `{@inheritDoc}` as the only thing in the JavaDoc. This will just copy it from the parent.

File: src/main/java/inf112/isolasjonsteamet/roborally/gui/screens/GameScreen.java [hunk 6]
```diff
@@ -2,15 +2,15 @@
...
+import inf112.isolasjonsteamet.roborally.app.RoboRallyGame;
 import inf112.isolasjonsteamet.roborally.gui.DelegatingInputProcessor;
 
 /**
  * A screen which wraps the game itself.
  */
 public class GameScreen implements Screen, DelegatingInputProcessor {
 
-	private final HelloWorld helloWorld = new HelloWorld();
+	private final RoboRallyGame helloWorld = new RoboRallyGame();
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/gui/screens/GameScreen.java]: Could use a better variable name here

**Review by Katrix — APPROVED** [review]

**Review by nooralindeflaten — APPROVED** [review]

**Pull request merged**